extensions/query-history/src/main.ts

```diff
@@ -5,37 +5,89 @@
+		// when it was clicked to compare, then if a click happens on the same element quickly enough we trigger the configured action
+		const clickTime = new Date().getTime();
+		if (lastSelectedItem.item === selectedItem && lastSelectedItem.time && (clickTime - lastSelectedItem.time) < DOUBLE_CLICK_TIMEOUT_MS) {
+			const doubleClickAction = vscode.workspace.getConfiguration(QUERY_HISTORY_CONFIG_SECTION).get<string>(DOUBLE_CLICK_ACTION_CONFIG_SECTION);
+			switch (doubleClickAction) {
+				case 'run':
+					await runQuery(selectedItem);
+					break;
+				case 'open':
+				default:
+					await openQuery(selectedItem);
+					break;
+			}
+			// Clear out the last selected item so we don't run the command again on a 3rd click
+			lastSelectedItem = {
+				item: undefined,
+				time: undefined
+			};
+		} else {
+			// Update the last selected item since we didn't run a command
+			lastSelectedItem = {
+				item: selectedItem,
+				time: clickTime
+			};
+		}
+
+	}));
 	context.subscriptions.push(vscode.commands.registerCommand('queryHistory.open', async (item: QueryHistoryItem) => {
-		return azdata.queryeditor.openQueryDocument(
-			{
-				content: item.queryText
-			}, item.connectionProfile?.providerId);
+		return openQuery(item);
 	}));
 	context.subscriptions.push(vscode.commands.registerCommand('queryHistory.run', async (item: QueryHistoryItem) => {
-		const doc = await azdata.queryeditor.openQueryDocument(
-			{
-				content: item.queryText
-			}, item.connectionProfile?.providerId);
-		await azdata.queryeditor.connect(doc.uri, item.connectionProfile?.connectionId || '');
-		azdata.queryeditor.runQuery(doc.uri);
+		return runQuery(item);
 	}));
 	context.subscriptions.push(vscode.commands.registerCommand('queryHistory.delete', (item: QueryHistoryItem) => {
-		provider.deleteItem(item);
+		treeDataProvider.deleteItem(item);
 	}));
 	context.subscriptions.push(vscode.commands.registerCommand('queryHistory.clear', () => {
-		provider.clearAll();
+		treeDataProvider.clearAll();
 	}));
 	context.subscriptions.push(vscode.commands.registerCommand('queryHistory.disableCapture', async () => {
-		return provider.setCaptureEnabled(false);
+		return treeDataProvider.setCaptureEnabled(false);
 	}));
 	context.subscriptions.push(vscode.commands.registerCommand('queryHistory.enableCapture', async () => {
-		return provider.setCaptureEnabled(true);
+		return treeDataProvider.setCaptureEnabled(true);
 	}));
 }
+
+async function openQuery(item: QueryHistoryItem): Promise<void> {
+	await azdata.queryeditor.openQueryDocument(
+		{
+			content: item.queryText
+		}, item.connectionProfile?.providerId);
+}
+
+async function runQuery(item: QueryHistoryItem): Promise<void> {
+	const doc = await azdata.queryeditor.openQueryDocument(
+		{
+			content: item.queryText
+		}, item.connectionProfile?.providerId);
+	await azdata.queryeditor.connect(doc.uri, item.connectionProfile?.connectionId || '');
```

Comment:
what will be the experience if the connection is not saved and already got disconnected? is this going to throw some weird error? I'd expect the connection dialog will open for user to provide the connection information, after the connection is made, the query should be executed.

Reply:
Yeah, it'll open the connection dialog and then execute the query afterwards.